### PR TITLE
feat: Add 'isTransactional' attribute.

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITE2ETracingTest.java [hunk 1]
```diff
@@ -241,7 +241,7 @@ private boolean dfsContainsCallStack(long spanId, List<String> expectedCallStack
 
   private static final int NUM_SPAN_ID_BYTES = 16;
 
-  private static final int GET_TRACE_RETRY_COUNT = 15;
+  private static final int GET_TRACE_RETRY_COUNT = 60;
```
Comment (jimit-j-shah) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITE2ETracingTest.java:244]: Was 15 not enough?

1 qps of retries. Max allowed is 5 qps. https://cloud.google.com/trace/docs/quotas#:~:text=Trace%20API%20usage%20limits,-API%20method&text=You%20have%20a%20total%20of,API%20calls%20per%2060%20seconds.

Comment (ehsannas) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITE2ETracingTest.java:244]: no it wasn't enough. our backoff is 1 second. so our max qps should be 1.

**Comment by jimit-j-shah** [comment]
Did you test this - as in how this shows up in the trace information? Please also update the [Trace catalog.](https://docs.google.com/document/d/1aCA--hwv38cWHmzdci2GCKxnCMZK8zyQ6KaasDsonwM/edit?resourcekey=0-ef9rl-JJ4umAnaoZoYTRrA&tab=t.0#heading=h.y73r5y1e7mc)

**Review by jimit-j-shah — APPROVED** [review]

**Comment by ehsannas** [comment]
> Did you test this - as in how this shows up in the trace information?

Yes. I also manually tested and observed the attribute in Cloud Trace.

>Please also update the Trace catalog

Done

**Pull request merged**